providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/controller/CacheController.java

```diff
@@ -0,0 +1,57 @@
+package dev.openfeature.contrib.providers.gofeatureflag.controller;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import dev.openfeature.contrib.providers.gofeatureflag.GoFeatureFlagProviderOptions;
+import dev.openfeature.contrib.providers.gofeatureflag.bean.BeanUtils;
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.ProviderEvaluation;
+import lombok.Builder;
+
+import java.time.Duration;
+
+/**
+ * CacheController is a controller to manage the cache of the provider.
+ */
+public class CacheController {
+    public static final long DEFAULT_CACHE_TTL_MS = 5L * 60L * 1000L;
+    public static final int DEFAULT_CACHE_CONCURRENCY_LEVEL = 1;
+    public static final int DEFAULT_CACHE_INITIAL_CAPACITY = 100;
+    public static final int DEFAULT_CACHE_MAXIMUM_SIZE = 100000;
+    private final Cache<String, ProviderEvaluation<?>> cache;
```

Comment:
As this introduce cache refactor, I think we may consider replacing it with [Caffeine](https://github.com/ben-manes/caffeine?tab=readme-ov-file#cache), as it is [Discouraged](https://javadoc.io/doc/com.google.guava/guava/latest/com/google/common/cache/package-summary.html) - [Discouraged](https://guava.dev/CacheBuilder) (in favor of [Caffeine](https://github.com/ben-manes/caffeine/wiki)) caching utilities.
It can be done in this PR or a new PR, by you or me. Considering it may introduce breaking changes at provider config, and change some cache config fields.

Reply:
I am going to pass on this one because I want to avoid adding breaking changes for now in the provider and his configuration.

Since I still like the idea, I have opened an issue https://github.com/thomaspoignant/go-feature-flag/issues/1985 to be sure that we do not forget it in the future.
But I wanted to have more breaking changes before going to `1.0.0`.